Feat/v8

andy9a9 opened the pull request:
- Switch to custom [kiutils](https://github.com/andy9a9/kiutils) fork due to lack of v8 features

Closes [!2](https://github.com/Mikrozone-Server/KiCad-to-csv/issues/2)

andy9a9 committed:
chore(exception): raise exception when happens

andy9a9 commented:
@EdizonTN Could you please verify if basic functionality for `v8` works?

PetoJanis commented:
**Export sch symbols** ...... Perfect
**Export pcb footprints** ......... OK, but some message was catched: 
"_Unable to process c:\Users\peter.janis\Documents\Workspace-KiCad8\LIB\PCB\noPCB_TerminalBlockPluggable_Phoenix.pretty\FK-MCP 1.5-10-STF-3.81.kicad_mod", err: Dimensions are not yet handled! Please report this bug along with the file being parsed._"
Src file: [https://github.com/Z-Elektrik/LIB/tree/main/PCB/THD_Terminal_Phoenix.pretty](https://github.com/Z-Elektrik/LIB/tree/main/PCB/THD_Terminal_Phoenix.pretty)

**Import sch symbols** ...... not good:
 not visible fields (Show attribute was not checked) will be set as visible after import
![image](https://github.com/user-attachments/assets/36b84a90-9511-4ab5-891c-5b6976c1f32f)
Check image. After export and import of the this element, all marked attributes are "checked"
Library you can find in the attachment: [Test.zip](https://github.com/user-attachments/files/16362307/Test.zip)

**Import pcb symbols** ...... looks OK

Tested in KiCad version 8,0.4

andy9a9 commented:
Ok, I see the problem... . We're processing the values of properties only. Means, if the property looks like:
```
(property "Footprint" "Generic:TestFootprint"
	(at 3.302 11.938 0)
	(effects
		(font
			(size 1.27 1.27)
		)
		(hide yes)
	)
)
```
we just get `Footprint`= `Generic:TestFootprint` all other entries are ignored. If we want to "extend" the export mechanism, we have to refactor the whole system and store the entries into csv like: `"[Footprint(value='Generic:TestFootprint', pos=Coordinate(X=3.302, Y=11.938, Z=0), ...]"`. Not sure if we wanted to go with this direction... .

andy9a9 committed:
feat(kiutils): generator_version

- switch to custom kiutils version for "generator_version" handling
 - kicad v8 has an "generator_version" property
  - use this information for different properties handling

andy9a9 commented:
@PetoJanis Could you please re-test it again? It's necessary to remove the previous `kiutils` version and install the one one from `requirements.txt`

andy9a9 commented:
@PetoJanis @EdizonTN I know that's exactly one year ago, but is it still relevant?

PetoJanis commented:
> @PetoJanis @EdizonTN I know that's exactly one year ago, but is it still relevant?

Relevant - I don't use the sch symbol import because it still sets all attributes to visible, which is wrong in practice. If it worked correctly, I would use it.
(tested with kiutils-1.4.8)

andy9a9 commented:
> Relevant - I don't use the sch symbol import because it still sets all attributes to visible, which is wrong in practice. If it worked correctly, I would use it. (tested with kiutils-1.4.8)

Yes but, did you test it according to my previous comment? You have to uninstall the official version and install it from requirements. If you're satisfied with this variant, I would say we can merge it. If not describe what should be changed.

PetoJanis commented:
Yes.
Check output log:
test symbol lib you can find in the attachment (created using KiCAD v9.0.7)
```
c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver>python -m pip install -r requirements.txt
Collecting kiutils@ https://github.com/andy9a9/kiutils/releases/download/v1.4.8/kiutils-1.4.8-py3-none-any.whl (from -r requirements.txt (line 1))
  Downloading https://github.com/andy9a9/kiutils/releases/download/v1.4.8/kiutils-1.4.8-py3-none-any.whl (93 kB)

c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver>Export-from-sch9_test.bat

c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver>python kicad-from-to-csv.py -a export c:\Users\petoj\Documents\Projects\Workspace-KiCad\PRJ\test\  c:\Users\petoj\Documents\Projects\Workspace-KiCad\PRJ\test\Listing.csv  1>>Listing.txt
Traceback (most recent call last):
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 443, in <module>
    Symbols(kicad_dirfile)
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 147, in __init__
    if (not self.generator_version or self.generator_version != "8.0")
            ^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'Symbols' object has no attribute 'generator_version'
```
[Test.zip](https://github.com/user-attachments/files/24449148/Test.zip)

andy9a9 committed:
feat(version): make the script compatible with versions higher than 8.0

andy9a9 commented:
> ```
>   File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 147, in __init__
>     if (not self.generator_version or self.generator_version != "8.0")
>             ^^^^^^^^^^^^^^^^^^^^^^
> AttributeError: 'Symbols' object has no attribute 'generator_version'
> ```

This was designed for version `=< 8.0`. So I made it compatible with `9.0`, pls test it again.

PetoJanis commented:
Ok, **export works fine**. Import too.
**But, after import** all attributes are set as **hidden** (see image). Refrence and Value was set as Show.
Now I realized that if we want to do a reverse import, we will also need to save all attribute parameters to the csv (i.e. Show, Show Name, H Align, V Align, Italic and Bold).
I don't know if it's a lot of work, but without this import will not be meaningful for working with live libraries. It will always overwrite the set attribute properties.
I suggest ending this branch as an export only and possibly creating a new one where there would be a different file format also (something Excel-like) due to the large number of fields.

<img width="1126" height="875" alt="2026-01-06 20_40_06-Test_B3S-1000, SMD 6x6mm, Switch Tactile, 50mA, 12V, Upthrow 0 5mm, Vertical, Om" src="https://github.com/user-attachments/assets/23331ead-c96e-47c5-9c12-1617941b1573" />

andy9a9 commented:
> Ok, **export works fine**. Import too. **But, after import** all attributes are set as **hidden**

That's obvious :)
Look at the import file what you provided...
```
(symbol "B3S-1000, SMD 6x6mm, Switch Tactile, 50mA, 12V, Upthrow 0.5mm, Vertical, Omron"
		(pin_names
			(offset 0)
		)
		(exclude_from_sim no)
		(in_bom yes)
		(on_board yes)
		(property "Reference" "S"
			(at 0 3.81 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
		(property "Value" "B3S-1000"
			(at 0 -5.715 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
		(property "Footprint" "SMD_Switch_Omron:SW_PUSH_450P600X600H430-4N_Omron"
			(at 0 8.89 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
		(property "Datasheet" "https://www.tme.eu/Document/1a744fb026617d7259891fb0eea6ebae/B3S.pdf"
			(at 0 6.35 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
		(property "Description" "Switch Tactile, SPSTNO, 6x6mm, 50mA, 24VDC, upthrow 0.5mm, vertical, B3F Serie"
			(at 0 0 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
		(property "Technology" "SMD"
			(at 0 11.43 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
		(property "MFG" "Omron"
			(at 0 34.29 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
		(property "MPN" "B3S-1000(P)"
			(at 0 19.05 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes) <--- should be hidden
			)
		)
```
Basically all of the properties have the `hide` flag set to `yes`. So they are exported into csv as hidden and imported into kicad with this value for sure.

PetoJanis commented:
I was retesting again.
Results:

1. export symbols - script doesn't read "generator_version" corrwctly. Still return attribute "ki_description" in file created with version 9.0.
```
(
                "ki_description"
                if (KiCadVersion(getattr(self, 'generator_version', None)) < "8.0")
                else "Description"
            ),
```
My temporary solution: Line 181-185 I was mark as comment and add line  "Description", as attribute for next tests.
 
2. I created 2 zip file "After_export.ZIP" and "After_import.ZIP" First one is library and exported csv file.
Second one is library after back import the csv file (in the csv I manually fill comment attributes).
Back import fully break the library file (missing header, symbol symbol B3S-1000... are created multiply, and all attributes fields are visible again).
In the back import, I add "-d" switch for print debug messages:
```
c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver>python kicad-from-to-csv.py -d -a import c:\Users\petoj\Documents\Projects\Workspace-KiCad\LIB\  c:\Users\petoj\Documents\Projects\Workspace-KiCad\LIB\Listing.csv  1>>Result.txt
Processing...
--- Logging error ---
Traceback (most recent call last):
  File "C:\Python\Python312\Lib\logging\__init__.py", line 1160, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 703, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 392, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 504, in <module>
    comp.export() if action_type_export else comp.update()
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 328, in update
    Symbols(first_component[0])
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 180, in __init__
    LOGGER.debug(f'"Generator"', getattr(self, 'generator_version', None))
Message: '"Generator"'
Arguments: (None,)
[
 c:\Users\petoj\Documents\Projects\Workspace-KiCad\LIB = [
--- Logging error ---
Traceback (most recent call last):
  File "C:\Python\Python312\Lib\logging\__init__.py", line 1160, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 703, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 392, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 504, in <module>
    comp.export() if action_type_export else comp.update()
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 358, in update
    self._sim_or_foot.load(l[0])
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 233, in load
    self.__dict__.update(obj.from_file(path).__dict__)
  File "C:\Python\Python312\Lib\site-packages\kiutils\symbol.py", line 523, in from_file
    item = cls.from_sexpr(sexpr.parse_sexp(infile.read()))
  File "C:\Python\Python312\Lib\site-packages\kiutils\symbol.py", line 547, in from_sexpr
    object = cls()
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 180, in __init__
    LOGGER.debug(f'"Generator"', getattr(self, 'generator_version', None))
Message: '"Generator"'
Arguments: (None,)
  AO3414, SOT23, N-Mosfet, 20V, 3A, AO Semi = {
   "Description": "Transistor MOSFET N-Channel, SMD, 20V, 3A, 85m, Vgs(th) 1.0V"
   "ki_keywords": "T"
   "Reference": "Q"
   "Value": "AO3414"
   "Footprint": "SMD_Semiconductor_Generic:SOT95P280X125-3N_SOT23-3"
   "Datasheet": "https://www.aosmd.com/sites/default/files/res/datasheets/AO3414.pdf"
   "Technology": "SMD"
   "MFG": "Alpha&Omega Semiconductor"
   "MPN": "AO3414"
   "Device_Marking": "AE"
   "OC_LCSC": "C48850"
   "OC_MOUSER": ""
   "OC_RS": ""
   "OC_DISTRELEC": ""
   "OC_TME": "AO3414"
   "OC_FARNELL": ""
   "OC_DIGIKEY": ""
   "OC_SOS": ""
   "Z-SYSCODE": "12354"
   "Assembly_Note": ""
   "Comment": "comment1"
   "ki_fp_filters": "SOT95P280X125-3?_SOT23-3__123"
  },
 ],
 c:\Users\petoj\Documents\Projects\Workspace-KiCad\LIB = [
--- Logging error ---
Traceback (most recent call last):
  File "C:\Python\Python312\Lib\logging\__init__.py", line 1160, in emit
    msg = self.format(record)
          ^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 999, in format
    return fmt.format(record)
           ^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 703, in format
    record.message = record.getMessage()
                     ^^^^^^^^^^^^^^^^^^^
  File "C:\Python\Python312\Lib\logging\__init__.py", line 392, in getMessage
    msg = msg % self.args
          ~~~~^~~~~~~~~~~
TypeError: not all arguments converted during string formatting
Call stack:
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 504, in <module>
    comp.export() if action_type_export else comp.update()
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 358, in update
    self._sim_or_foot.load(l[0])
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 233, in load
    self.__dict__.update(obj.from_file(path).__dict__)
  File "C:\Python\Python312\Lib\site-packages\kiutils\symbol.py", line 523, in from_file
    item = cls.from_sexpr(sexpr.parse_sexp(infile.read()))
  File "C:\Python\Python312\Lib\site-packages\kiutils\symbol.py", line 547, in from_sexpr
    object = cls()
  File "c:\Users\petoj\Documents\Projects\Workspace-KiCad\scripting\KiCad-to-csv-feat-v9_gen_ver\kicad-from-to-csv.py", line 180, in __init__
    LOGGER.debug(f'"Generator"', getattr(self, 'generator_version', None))
Message: '"Generator"'
Arguments: (None,)
  B3S-1000, SMD 6x6mm, Switch Tactile, 50mA, 12V, Upthrow 0.5mm, Vertical, Omron = {
   "Description": "Switch Tactile, SPSTNO, 6x6mm, 50mA, 24VDC, upthrow 0.5mm, vertical, B3F Serie"
   "ki_keywords": "Switch,Button"
   "Reference": "S"
   "Value": "B3S-1000"
   "Footprint": "SMD_Switch_Omron:SW_PUSH_450P600X600H430-4N_Omron"
   "Datasheet": "https://www.tme.eu/Document/1a744fb026617d7259891fb0eea6ebae/B3S.pdf"
   "Technology": "SMD"
   "MFG": "Omron"
   "MPN": "B3S-1000(P)"
   "Device_Marking": ""
   "OC_LCSC": "C180420"
   "OC_MOUSER": "653-B3S-1000"
   "OC_RS": "183-701"
   "OC_DISTRELEC": "135-59-087"
   "OC_TME": "B3S-1000P"
   "OC_FARNELL": "177807"
   "OC_DIGIKEY": "SW415-ND"
   "OC_SOS": "6656"
   "Z-SYSCODE": "905"
   "Assembly_Note": ""
   "Comment": "comment2"
"Unable to process c:\Users\petoj\Documents\Projects\Workspace-KiCad\LIB\Test.kicad_sym", err: 'ki_fp_filters'
]
Done
```
[After_export.ZIP](https://github.com/user-attachments/files/24483897/After_export.ZIP)
[After_Import.ZIP](https://github.com/user-attachments/files/24483898/After_Import.ZIP)

andy9a9 commented:
> export symbols - script doesn't read "generator_version" corrwctly. Still return attribute "ki_description" in file created with version 9.0.
> 
> ```
> (
>                 "ki_description"
>                 if (KiCadVersion(getattr(self, 'generator_version', None)) < "8.0")
>                 else "Description"
>             ),
> ```

I re-tested it multiple times with your input files:
```
(kicad_symbol_lib
	(version 20241209)
	(generator "kicad_symbol_editor")
	(generator_version "9.0")
	(symbol "AO3414, SOT23, N-Mosfet, 20V, 3A, AO Semi"
		(pin_numbers
			(hide yes)
		)
		(pin_names
			(offset 1.016)
			(hide yes)
		)
		(exclude_from_sim no)
		(in_bom yes)
		(on_board yes)
		(property "Reference" "Q"
			(at 0 5.08 0)
			(effects
				(font
					(size 1.27 1.27)
				)
			)
		)
		(property "Value" "AO3414"
			(at 0 -5.08 0)
			(effects
				(font
					(size 1.27 1.27)
				)
			)
		)
		(property "Footprint" "SMD_Semiconductor_Generic:SOT95P280X125-3N_SOT23-3"
			(at 1.27 6.35 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes)
			)
		)
		(property "Datasheet" "https://www.aosmd.com/sites/default/files/res/datasheets/AO3414.pdf"
			(at 0 7.62 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes)
			)
		)
		(property "Description" "Transistor MOSFET N-Channel, SMD, 20V, 3A, 85mΩ, Vgs(th) 1.0V"
			(at 0 0 0)
			(effects
				(font
					(size 1.27 1.27)
				)
				(hide yes)
			)
		)
...
```
```sh
python3 kicad-from-to-csv.py -d -a export tests/symbols/Test9.kicad_sym out.csv
Processing...
[
 tests/symbols/Test9.kicad_sym = [
  AO3414, SOT23, N-Mosfet, 20V, 3A, AO Semi = {
   "Description": "Transistor MOSFET N-Channel, SMD, 20V, 3A, 85mΩ, Vgs(th) 1.0V"
   "ki_keywords": "T"
   "Reference": "Q"
   "Value": "AO3414"
   "Footprint": "SMD_Semiconductor_Generic:SOT95P280X125-3N_SOT23-3"
   "Datasheet": "https://www.aosmd.com/sites/default/files/res/datasheets/AO3414.pdf"
   "Technology": "SMD"
   "MFG": "Alpha&Omega Semiconductor"
   "MPN": "AO3414"
   "Device_Marking": "AE"
   "OC_LCSC": "C48850"
   "OC_MOUSER": ""
   "OC_RS": ""
   "OC_DISTRELEC": ""
   "OC_TME": "AO3414"
   "OC_FARNELL": ""
   "OC_DIGIKEY": ""
   "OC_SOS": ""
   "Z-SYSCODE": "12354"
   "Assembly_Note": ""
   "Comment": ""
   "ki_fp_filters": "SOT95P280X125-3?_SOT23-3__123"
  },
  B3S-1000, SMD 6x6mm, Switch Tactile, 50mA, 12V, Upthrow 0.5mm, Vertical, Omron = {
   "Description": "Switch Tactile, SPSTNO, 6x6mm, 50mA, 24VDC, upthrow 0.5mm, vertical, B3F Serie"
   "ki_keywords": "Switch,Button"
   "Reference": "S"
   "Value": "B3S-1000"
   "Footprint": "SMD_Switch_Omron:SW_PUSH_450P600X600H430-4N_Omron"
   "Datasheet": "https://www.tme.eu/Document/1a744fb026617d7259891fb0eea6ebae/B3S.pdf"
   "Technology": "SMD"
   "MFG": "Omron"
   "MPN": "B3S-1000(P)"
   "Device_Marking": ""
   "OC_LCSC": "C180420"
   "OC_MOUSER": "653-B3S-1000"
   "OC_RS": "183-701"
   "OC_DISTRELEC": "135-59-087"
   "OC_TME": "B3S-1000P"
   "OC_FARNELL": "177807"
   "OC_DIGIKEY": "SW415-ND"
   "OC_SOS": "6656"
   "Z-SYSCODE": "905"
   "Assembly_Note": ""
   "Comment": ""
   "ki_fp_filters": ""
Test9.kicad_sym:B3S-1000, SMD 6x6mm, Switch Tactile, 50mA, 12V, Upthrow 0.5mm, Vertical, Omron: "Label" is missing
Test9.kicad_sym:B3S-1000, SMD 6x6mm, Switch Tactile, 50mA, 12V, Upthrow 0.5mm, Vertical, Omron: "ki_fp_filters" is missing
  },
 ],
]
Done
```
and the output file:
```
"Lib_PATH+FILENAME","COMPONENT","Description","ki_keywords","Reference","Value","Footprint","Datasheet","Technology","MFG","MPN","Device_Marking","OC_LCSC","OC_MOUSER","OC_RS","OC_DISTRELEC","OC_TME","OC_FARNELL","OC_DIGIKEY","OC_SOS","Z-SYSCODE","Assembly_Note","Comment","ki_fp_filters"
"tests/symbols/Test9.kicad_sym","AO3414, SOT23, N-Mosfet, 20V, 3A, AO Semi","Transistor MOSFET N-Channel, SMD, 20V, 3A, 85mΩ, Vgs(th) 1.0V","T","Q","AO3414","SMD_Semiconductor_Generic:SOT95P280X125-3N_SOT23-3","https://www.aosmd.com/sites/default/files/res/datasheets/AO3414.pdf","SMD","Alpha&Omega Semiconductor","AO3414","AE","C48850","","","","AO3414","","","","12354","","","SOT95P280X125-3?_SOT23-3__123"
"tests/symbols/Test9.kicad_sym","B3S-1000, SMD 6x6mm, Switch Tactile, 50mA, 12V, Upthrow 0.5mm, Vertical, Omron","Switch Tactile, SPSTNO, 6x6mm, 50mA, 24VDC, upthrow 0.5mm, vertical, B3F Serie","Switch,Button","S","B3S-1000","SMD_Switch_Omron:SW_PUSH_450P600X600H430-4N_Omron","https://www.tme.eu/Document/1a744fb026617d7259891fb0eea6ebae/B3S.pdf","SMD","Omron","B3S-1000(P)","","C180420","653-B3S-1000","183-701","135-59-087","B3S-1000P","177807","SW415-ND","6656","905","","",""
```
As and you can see **Description** was exported correctly. It has to be working fine, while I tested it on multiple environments (Linux, MAC and Windows). So I maybe you're using a wrong `kiutils` version from global environment and not a local from `.venv`.
> I created 2 zip file "After_export.ZIP" and "After_import.ZIP" First one is library and exported csv file.

This is not really clear what you wanted to achieve. Please describe it more properly with exact commands. You exported some symbols/footprint (which one and from which KiCad version) and wanted to import it back (to which version)? Did you modify something meanwhile? But take a look on [comment](https://github.com/Mikrozone-Server/KiCad-to-csv/pull/4#issuecomment-2258896659) which I wrote some year ago. There is basically no "nice" way to export everything... . That's the reason why we're taking care only about some stuff.

andy9a9 commented:
@PetoJanis @EdizonTN As you're basically ingornig my responses... 🤣, nevertheless, does it still make sense when https://forum.kicad.info/t/post-v9-new-features-and-development-news/58848/20 was implemented? Basically you can modify fields on each symbol in library just at once. Or is that something still missing?

PetoJanis commented:
Sorry, I was waiting for v10 :-) . 
No, I have forgot :-( . 
OK, new mass edit in v10 is helpful but is working only to one library not more than one.
Second limitation - not existing csv export/import (needs for other ERP system connection/synchronization)

Ok, we can close all issues and move to v10 compatibility check.

andy9a9 commented:
> OK, new mass edit in v10 is helpful but is working only to one library not more than one.

OK, but why you need an option to edit more than 1 library at once?

> Second limitation - not existing csv export/import (needs for other ERP system connection/synchronization)

So you can't load (in normal way) the old kicad library to be compatible with new version? I guess it has to be done automatically, or? Or why you need the CSV import/export function then?

> Ok, we can close all issues and move to v10 compatibility check.

What does it mean? All these changes which I did have to be throw away?

I'm just collecting the features which could be potentially added directly to kicad.

PetoJanis commented:
> > OK, new mass edit in v10 is helpful but is working only to one library not more than one.
> 
> OK, but why you need an option to edit more than 1 library at once?

At 90% I need to export all symbols from all libraries to CSV, then to Excel and sync with our ERP system.
At 10% I need to add a new field to all symbols in all libraries with some default values ​​(eg Status: NEW/CHECKED/NRND).
It depends on the scenario. 
[Public example](https://docs.google.com/spreadsheets/d/176wO_S8_nby6Lyd8MGY-Abj0bwKNF8hLd7M2Ny2Cflg/edit?gid=778825821#gid=778825821
)



> > Second limitation - not existing csv export/import (needs for other ERP system connection/synchronization)
> 
> So you can't load (in normal way) the old kicad library to be compatible with new version? I guess it has to be done 
>automatically, or? Or why you need the CSV import/export function then?

KiCad v9 libraries are updated to the new v10 version after the first save.
I have not tested this script with the new KiCad v10. I don't know if it will work properly.



> > Ok, we can close all issues and move to v10 compatibility check.
> 
> What does it mean? All these changes which I did have to be throw away?

I think not so generally. But for version 9 it doesn't matter anymore. All further tests and updates should be done with version 10 only.

> I'm just collecting the features which could be potentially added directly to kicad.

Do you want to add it directly to KiCad? That would be good!

andy9a9 commented:
Superseded by #5
